rust_sodium-sys: Version change to 0.10.4

lionel-faber opened the pull request:
- download libsodium binaries from s3
- update file hashes for the new binaries
- bump version

S-Coyle reviewed (dismissed):
Checked SHA256 checksums against downloads - all matches as it should

pierrechevalier83 reviewed (dismissed):
Thanks for that Lionel 😀 I hit the exact same issue today 😁

lionel-faber committed:
rust_sodium-sys: Version change to 0.10.4

- download libsodium binaries from s3
- update file hashes for the new binaries
- bump version
- clear appveyor cache when Cargo.toml changes

lionel-faber commented:
Closing this as #107 upgrades libsodium 1.0.17
